fab/operations/offline.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from fabric.api import env, local
+from fab.utils import generate_bower_command
+from fab.const import (
+    OFFLINE_STAGING_DIR,
+    WHEELS_ZIP_NAME,
+    HQ_ZIP_NAME,
+    BOWER_ZIP_NAME,
+    NPM_ZIP_NAME,
+)
+
+
+def prepare_zipfiles():
+    local(
+        'git clone --depth 1 --recursive https://github.com/dimagi/commcare-hq.git {}/commcare-hq'
+        .format(OFFLINE_STAGING_DIR)
+    )
+
+    # After we've created our HQ code zip, let's get bower and npm
```

Comment:
no this is definitely a step which i'm least confident in. it's very intensive process to attempt to clone the whole repo and then upload and/or push to each server. open to new ideas